var/spack/repos/builtin/packages/py-pyprecice/package.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyPyprecice(PythonPackage):
+    """
+    This package provides python language bindings for the
+    C++ library preCICE.
+    """
+
+    homepage = "https://www.precice.org"
+    git = "https://github.com/precice/python-bindings.git"
+    url = "https://github.com/precice/python-bindings/archive/v2.0.0.1.tar.gz"
+    maintainers = ["ajaust", "BenjaminRueth"]
+
+    # Always prefer final version of release candidate
+    version("develop", branch="develop")
+    version(
+        "2.1.1.1",
+        preferred=True,
+        sha256="972f574549344b6155a8dd415b6d82512e00fa154ca25ae7e36b68d4d2ed2cf4",
+    )
+    version(
+        "2.1.1.1rc2",
+        sha256="beefd2bfab0e9eca6a8f58168d650611193a567479237db0e18a4ca0783269eb",
+    )
+    version(
+        "2.1.1.1rc1",
+        sha256="2fb4ed1a6c12bb3a093d5eac30740a8d0fc61e86bfadbfa37a12e0794059fee5",
+    )
+    version(
+        "2.1.0.1",
+        preferred=True,
+        sha256="ac5cb7412c6b96b08a04fa86ea38e52d91ea739a3bd1c209baa93a8275e4e01a",
+    )
```

Comment:
Multiple versions can't be preferred. Also, it's fine to put each version on a single line, we ignore checksums in our flake8 line-length checks.

Reply:
Fixed.